trino/integration_test.go

```diff
@@ -1325,3 +1460,240 @@ func TestIntegrationLargeQuery(t *testing.T) {
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM tpch.sf1.customer LIMIT 1000", sql.Named("X-Trino-Query-Data-Encoding", "json+zstd"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	count := 0
+	for rows.Next() {
+		count++
+		var col tpchRow
+		err = rows.Scan(
+			&col.CustKey,
+			&col.Name,
+			&col.Address,
+			&col.NationKey,
+			&col.Phone,
+			&col.AcctBal,
+			&col.MktSegment,
+			&col.Comment,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	if rows.Err() != nil {
+		t.Fatal(err)
+	}
+	if count != 1000 {
+		t.Fatal("not enough rows returned:", count)
+	}
+}
+
+func TestIntegrationSelectTpch1000SpoolingSpolledSegmentJsonEncoded(t *testing.T) {
+	db := integrationOpen(t)
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM tpch.sf1.customer LIMIT 1000", sql.Named("X-Trino-Query-Data-Encoding", "json"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	count := 0
+	for rows.Next() {
+		count++
+		var col tpchRow
+		err = rows.Scan(
+			&col.CustKey,
+			&col.Name,
+			&col.Address,
+			&col.NationKey,
+			&col.Phone,
+			&col.AcctBal,
+			&col.MktSegment,
+			&col.Comment,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	if rows.Err() != nil {
+		t.Fatal(err)
+	}
+	if count != 1000 {
+		t.Fatal("not enough rows returned:", count)
+	}
+}
+
+func TestIntegrationSelectTpch1000SpoolingSpolledSegmentJsonLz4Encoded(t *testing.T) {
+	db := integrationOpen(t)
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM tpch.sf1.customer LIMIT 1000", sql.Named("X-Trino-Query-Data-Encoding", "json+lz4"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	count := 0
+	for rows.Next() {
+		count++
+		var col tpchRow
+		err = rows.Scan(
+			&col.CustKey,
+			&col.Name,
+			&col.Address,
+			&col.NationKey,
+			&col.Phone,
+			&col.AcctBal,
+			&col.MktSegment,
+			&col.Comment,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	if rows.Err() != nil {
+		t.Fatal(err)
+	}
+	if count != 1000 {
+		t.Fatal("not enough rows returned:", count)
+	}
+}
+
+func TestIntegrationSelectTpch100SpoolingInlineSegmentJsonZstdEncoded(t *testing.T) {
+	db := integrationOpen(t)
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM tpch.sf1.customer LIMIT 100", sql.Named("X-Trino-Query-Data-Encoding", "json+zstd"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer rows.Close()
+	count := 0
+	for rows.Next() {
+		count++
+		var col tpchRow
+		err = rows.Scan(
+			&col.CustKey,
+			&col.Name,
+			&col.Address,
+			&col.NationKey,
+			&col.Phone,
+			&col.AcctBal,
+			&col.MktSegment,
+			&col.Comment,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	if rows.Err() != nil {
+		t.Fatal(err)
+	}
+	if count != 100 {
+		t.Fatal("not enough rows returned:", count)
+	}
+}
+
+func TestIntegrationSelectTpch100SpoolingInlineSegmentJsonlz4Encoded(t *testing.T) {
+	db := integrationOpen(t)
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM tpch.sf1.customer LIMIT 100", sql.Named("X-Trino-Query-Data-Encoding", "json+lz4"))
```

Comment:
We need a test that will make sure results are returned in order. Generate a sequence of numbers and assert that it's monotonic (every value is greater than the previous one).